app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostFragment.kt

```diff
@@ -201,6 +204,13 @@ class ZimHostFragment : BaseFragment(), ZimHostCallbacks, ZimHostContract.View {
     }
   }
 
+  private fun setBooksAdapter(booksList: List<BooksOnDiskListItem>?) {
```

Comment:
get rid of this method, it is a terrible liar. It does not set adapter and sometimes it cause a database load

Reply:
@macgills How should I set item in adapter?

Reply:
`booksAdapter.items= someList`

you don't need to notify dataset changed because the setter takes care of that

```
  var items: List<ITEM> = mutableListOf()
    set(value) {
      field = value
      notifyDataSetChanged()
    }
```

You also don't need to ever accept null in this method if you change your fields to be non nullable.

so this method becomes **drum roll**  `booksAdapter.items= someList` which doesn't need to be a separate method